examples/query/synth_data_llm_filter.py

```diff
@@ -0,0 +1,169 @@
+
+opensearch_port = args.osport
+
+os_client_args = {
+    "hosts": [{"host": opensearch_host, "port": opensearch_port}],
+    "http_compress": True,
+    "http_auth": ("admin", "admin"),
+    "use_ssl": True,
+    "verify_certs": False,
+    "ssl_assert_hostname": False,
+    "ssl_show_warn": False,
+    "timeout": 120,
+}
+
+os_client = OpenSearch(**os_client_args)  # type: ignore
+
+index_settings = {
+    "body": {
+        "settings": {"index.knn": True, "number_of_shards": 5, "number_of_replicas": 1},
+        "mappings": {
+            "properties": {
+                "embedding": {
+                    "dimension": 384,
+                    "method": {
+                        "engine": "faiss",
+                        "space_type": "l2",
+                        "name": "hnsw",
+                        "parameters": {},
+                    },
+                    "type": "knn_vector",
+                }
+            }
+        },
+    }
+}
+
+# The number of documents to generate.
+max_docs = int(args.maxdocs)
+
+# The maximum number of elements in each document.
+max_elems = int(args.maxelems)
+
+
+def make_element(text):
+    """
+    Creates an element with the text representation.
+
+    Args:
+        text (str): The text representation of the element.
+
+    Returns:
+        dict: A dictionary representing the element with the following keys:
+            - 'properties': A dictionary of properties associated with the element.
+            - 'text_representation': The text representation of the element.
+    """
+#    return Element(text_representation=text)
+    return {'properties': {}, 'text_representation': text}
+
+def generate_elements(keywords, probabilities, max_elements):
+
+    strings = []
+    num = random.randint(1, max_elements)
+    base_text = """
+    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed auctor in enim sit amet tristique. 
+    In eu egestas ex. Maecenas porttitor gravida libero ac porttitor. Morbi elementum eleifend nisl.
+     Nullam auctor magna nec mollis ultricies. Mauris non pulvinar turpis. Pellentesque dictum 
+     tortor ac elit feugiat varius. Nulla sollicitudin, lectus non venenatis finibus, nulla ex 
+     tempor urna, id ornare est nisl et lacus. Pellentesque commodo turpis in nibh aliquet viverra. 
+     Curabitur varius, elit nec finibus scelerisque, nibh eros congue dui, et ultrices justo magna 
+     eu sem. Sed ultricies posuere posuere. Ut euismod neque at diam interdum, sed rhoncus quam 
+     iaculis. Morbi sit amet tempus orci, a dictum metus. Donec a imperdiet sem. Vestibulum tincidunt 
+     gravida turpis vitae aliquet.
+    """
+    for i in range(num):
+        text = "base_text" + str(i)
+        selected_keywords = ""
+        for keyword, probability in zip(keywords, probabilities):
+            if random.random() < probability:
+                selected_keywords = selected_keywords + " " + keyword
```

Comment:
fwiw, this is O(n); Probably doesn't matter, but the cleaner way would be to do it as a list. I think:
```
   " ".join([k for k in keyword_to_probability if random.random() < keyword_to_probability[k]])
```
does it in a single list comprehension.